src/pybind/mgr/prometheus/module.py

```diff
@@ -300,6 +300,8 @@ def get_metadata_and_osd_status(self):
             osd_metadata = self.get_metadata("osd", str(id_))
             dev_keys = ("backend_filestore_dev_node", "bluestore_bdev_dev_node")
             osd_dev_node = None
+            if not osd_metadata:
```

Comment:
Nit: Would you mind moving this test up two lines? No need to do two assignments before bailing out.